### PR TITLE
Crud

### PR DESCRIPTION
**Pull request by maartenvanasperen**

### CODE REVIEW

File: app/Resources/views/main.html.twig [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!doctype html>
```
Comment (TimGroenenberg) [inline app/Resources/views/main.html.twig]: Fix the indentation in this file.

File: app/Resources/views/pages/create.html.twig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{% extends "main.html.twig" %}
```
Comment (TimGroenenberg) [inline app/Resources/views/pages/create.html.twig]: Use single quotes.

File: app/Resources/views/pages/edit.html.twig [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{% extends "main.html.twig" %}
```
Comment (TimGroenenberg) [inline app/Resources/views/pages/edit.html.twig]: Use single quotes.
Also this view is the same as your create view. Can't you combine them?

File: app/Resources/views/pages/overview.html.twig [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{% extends "main.html.twig" %}
```
Comment (TimGroenenberg) [inline app/Resources/views/pages/overview.html.twig]: Use single quotes.

File: app/config/routing.yml [hunk 5]
```diff
@@ -2,6 +2,26 @@ app:
     resource: '@AppBundle/Controller/'
     type: annotation
 
+Race_overview:
```
Comment (TimGroenenberg) [inline app/config/routing.yml]: Use lowercase route names. Also your route paths are very generic at the moment. I would add something in the path so you know what you're adding/editing or deleting.

File: app/Resources/views/pages/overview.html.twig [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+{% block title %}
+    Overzicht van alle {{ races|length }} races
+{% endblock %}
+
+{% block body %}
+
```
Comment (TimGroenenberg) [inline app/Resources/views/pages/overview.html.twig]: Remove newline.

File: app/Resources/views/pages/overview.html.twig [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    <a href="{{ path('Race_create') }}">Nieuwe race toevoegen.</a>
+    <table>
+        <tr>
+            <th>Locatie</th>
+            <th>Beschrijving</th>
+            <th>Datum/tijd</th>
+            <th>Duur(in uren)</th>
+        </tr>
+        {% for race in races %}
```
Comment (TimGroenenberg) [inline app/Resources/views/pages/overview.html.twig]: You can add a check in your template to see if there are any races and display another message when there are none.

File: src/AppBundle/Controller/RaceController.php [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\Extension\Core\Type\TextareaType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\HttpFoundation\Request;
+
+class RaceController extends Controller
+{
+    public function indexAction (){
+        return $this->render('default/index.html.twig', [
+            'base_dir' => realpath($this->getParameter('kernel.project_dir')).DIRECTORY_SEPARATOR,
```
Comment (TimGroenenberg) [inline src/AppBundle/Controller/RaceController.php]: Why do you need this?

File: src/AppBundle/Controller/RaceController.php [hunk 9]
```diff
@@ -0,0 +1,101 @@
+<?php
```
Comment (TimGroenenberg) [inline src/AppBundle/Controller/RaceController.php]: Missing docblock for your class and all the functions in this file.

File: src/AppBundle/Controller/RaceController.php [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+class RaceController extends Controller
+{
+    public function indexAction (){
+        return $this->render('default/index.html.twig', [
+            'base_dir' => realpath($this->getParameter('kernel.project_dir')).DIRECTORY_SEPARATOR,
+        ]);
+    }
+
+    public function overviewAction(){
+        $er = $this->getDoctrine()->getRepository(Race::class);
```
Comment (TimGroenenberg) [inline src/AppBundle/Controller/RaceController.php]: `$er` is not a really self-explanatory name. Use something more obvious.

File: src/AppBundle/Controller/RaceController.php [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+
+        $em = $this->getDoctrine()->getManager();
+        $em->remove($race);
+        $em->flush();
+
+        return $this->redirectToRoute('Race_overview');
+    }
+
+    public function createRaceForm($race){
+        $form = $this->createFormBuilder($race)
```
Comment (TimGroenenberg) [inline src/AppBundle/Controller/RaceController.php]: You can use a Formtype for this see:
http://symfony.com/doc/3.4/best_practices/forms.html#building-forms

File: src/AppBundle/Entity/Result.php [hunk 12]
```diff
@@ -167,5 +167,33 @@ public function getDrivers()
     {
         return $this->drivers;
     }
-}
+    /**
+     * @var \AppBundle\Entity\Driver
+     */
+    private $driver;
+
+
```
Comment (TimGroenenberg) [inline src/AppBundle/Entity/Result.php]: Remove newline.

File: web/app_dev.php [hunk 13]
```diff
@@ -12,7 +12,7 @@
 // Feel free to remove this, extend it, or make something more sophisticated.
 if (isset($_SERVER['HTTP_CLIENT_IP'])
     || isset($_SERVER['HTTP_X_FORWARDED_FOR'])
-    || !(in_array(@$_SERVER['REMOTE_ADDR'], ['127.0.0.1', '::1'], true) || PHP_SAPI === 'cli-server')
+    || !(in_array(@$_SERVER['REMOTE_ADDR'], ['127.0.0.1','10.123.123.232', '::1'], true) || PHP_SAPI === 'cli-server')
```
Comment (TimGroenenberg) [inline web/app_dev.php]: Why is this added?

File: src/AppBundle/Tests/Controller/RaceControllerTest.php [hunk 14]
```diff
@@ -0,0 +1,9 @@
+<?php
```
Comment (TimGroenenberg) [inline src/AppBundle/Tests/Controller/RaceControllerTest.php]: If you don't plan on adding tests here you don't need to add the file now.

File: web/app_dev.php (hunk 13 above)
Comment (maartenvanasperen) [reply web/app_dev.php]: Cause I whitelisted the ip, now I can access:
http://formula1.virginia.choffice.nl/app_dev.php

File: src/AppBundle/Tests/Controller/RaceControllerTest.php (hunk 14 above)
Comment (maartenvanasperen) [reply src/AppBundle/Tests/Controller/RaceControllerTest.php]: I dont see this file in branch crud on my local machine

File: app/Resources/views/pages/overview.html.twig [hunk 15]
```diff
@@ -0,0 +1,32 @@
+{% extends 'main.html.twig' %}
+
+{% block title %}
+    Overzicht van alle {{ races|length }} races
+{% endblock %}
+
+{% block body %}
+    <a href="{{ path('race_create') }}">Nieuwe race toevoegen.</a><br>
+    {% if races is not empty %}
+    <table>
```
Comment (TimGroenenberg) [inline app/Resources/views/pages/overview.html.twig]: Indent this.

File: src/AppBundle/Controller/RaceController.php [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+            ]
+        );
+    }
+
+    /**
+     * A race form where the admin can create a new race
+     *
+     * @param Request $request
+     *
+     * @return Response|RedirectResponse|
```
Comment (TimGroenenberg) [inline src/AppBundle/Controller/RaceController.php]: Remove `|` at the end of the `@return`.

File: src/AppBundle/Controller/RaceController.php [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+        );
+    }
+    /**
+     * Functionality so the admin can delete a race
+     *
+     * @param integer $id
+     *
+     * @return RedirectResponse
+     */
+
```
Comment (TimGroenenberg) [inline src/AppBundle/Controller/RaceController.php]: Remove newline.
